### ISSUE TITLE
SDImageAPNGCoder crash in iOS 16.0.2

### ISSUE THREAD

**Issue by wenjingjie** [SDWebImage/SDWebImage#3418]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | 16.0.2
 SDWebImage Version      | 5.13.2
 Integration Method      | cocoapods
 Xcode Version           |  Xcode 13
 Repro rate              | only once
 Repro with our demo prj | no 
 Demo project link       | no

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

the stack trace is below:

```
[Fatal](url) Exception: NSInvalidArgumentException
-[SDImageAPNGCoder canDecodeFromData:]: unrecognized selector sent to instance 0x280a622e0
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x9db4 __exceptionPreprocess
1  libobjc.A.dylib                0x159a0 objc_exception_throw
2  CoreFoundation                 0x169c24 +[NSObject(NSObject) _copyDescription]
3  CoreFoundation                 0x1ed20 ___forwarding___
4  CoreFoundation                 0x8298c _CF_forwarding_prep_0
5  HLLDriver-SEA                  0x13d2600 -[SDImageCodersManager decodedImageWithData:options:] + 108 (SDImageCodersManager.m:108)
6  HLLDriver-SEA                  0x13cd774 SDImageCacheDecodeImageData + 78 (SDImageCacheDefine.m:78)
7  HLLDriver-SEA                  0x13caab8 -[SDImageCache diskImageForKey:data:options:context:] + 496 (SDImageCache.m:496)
8  HLLDriver-SEA                  0x13cb508 __73-[SDImageCache queryCacheOperationForKey:options:context:cacheType:done:]_block_invoke.297 + 634 (SDImageCache.m:634)
9  HLLDriver-SEA                  0x13cb6d8 __73-[SDImageCache queryCacheOperationForKey:options:context:cacheType:done:]_block_invoke.300 + 657 (SDImageCache.m:657)
10 libdispatch.dylib              0x63850 _dispatch_call_block_and_release
11 libdispatch.dylib              0x647c8 _dispatch_client_callout
12 libdispatch.dylib              0x3f854 _dispatch_lane_serial_drain$VARIANT$armv81
13 libdispatch.dylib              0x402e4 _dispatch_lane_invoke$VARIANT$armv81
14 libdispatch.dylib              0x4a000 _dispatch_workloop_worker_thread
15 libsystem_pthread.dylib        0x1b50 _pthread_wqthread
16 libsystem_pthread.dylib        0x167c start_wqthread
```

please help to check, thanks~~

**Comment by dreampiggy** [comment]
Is this reproducable ?

I check the SDWebImage demo, it has `apng` image url, and indeed call the`SDImageAPNGCoder`

Actually, from the code side, `SDImageAPNGCoder` is subclass of `SDImageAnimatedCoder` ,which implements the `canDecodeFromData:` function. So it should responds to that selector.

**Comment by wenjingjie** [comment]
> Is this reproducable ?
> 
> I check the SDWebImage demo, it has `apng` image url, and indeed call the`SDImageAPNGCoder`
> 
> Actually, from the code side, `SDImageAPNGCoder` is subclass of `SDImageAnimatedCoder` ,which implements the `canDecodeFromData:` function. So it should responds to that selector.

it's not reproducible. 

BTW the main thread stack trace of this crash is below

```
com.apple.main-thread
0  libFontParser.dylib            0x22c70 ApplyVariationsToGlyph(int, short*, unsigned int, glyphVariation const*, int, short const*, int, int, unsigned short const*, float*, float*, float*, float*, float*, float*, int) + 268
1  libFontParser.dylib            0x23708 ApplyFeaturesToOutline(fsg_SplineKey*, fnt_ElementType*) + 324
2  libFontParser.dylib            0x83814 StretchGlyph(fsg_SplineKey*, fnt_ElementType*, transformState const*, std::__1::unique_ptr<float [], std::__1::default_delete<float []> >*, unsigned char, unsigned char*, unsigned int) + 336
3  libFontParser.dylib            0x84c0c CreateGlyphOutline(fsg_SplineKey*, unsigned short, unsigned char) + 88
4  libFontParser.dylib            0x449bc CreateGlyphElement(fsg_SplineKey*, int, unsigned char, unsigned char) + 308
5  libFontParser.dylib            0x44b50 CreateScalerGlyphBlock(fsg_SplineKey*, memoryContext*, scalerGlyph const*) + 176
6  libFontParser.dylib            0x808f0 AssureGlyphBlock(fsg_SplineKey*, memoryContext*, scalerGlyph*) + 152
7  libFontParser.dylib            0x894fc TTRenderGlyphs + 536
8  libFontParser.dylib            0x1facc TConcreteFontScaler::CopyGlyphPath(unsigned short, CGAffineTransform const*) const + 352
9  libFontParser.dylib            0x7661c TFPFont::CopyGlyphPath(unsigned int) const + 248
10 libFontParser.dylib            0x6ac4 FPFontCopyGlyphPath + 36
11 CoreGraphics                   0x70d04 CGFontCreateGlyphPath + 44
12 CoreGraphics                   0x2afe8 CGGlyphBuilderLockBitmaps + 1156
13 CoreGraphics                   0x58644 render_glyphs + 248
14 CoreGraphics                   0x70c4c draw_glyph_bitmaps + 1128
15 CoreGraphics                   0x89398 ripc_DrawGlyphs + 1324
16 CoreText                       0xb7998 invocation function for block in DrawGlyphsAtPositions(TFont const*, unsigned short const*, CGPoint const*, unsigned long, CGContext*, (anonymous namespace)::DrawGlyphsConfiguration const&) + 224
17 CoreText                       0x343b0 EnumerateOverlappingGlyphs(CGContext*, TFont const&, unsigned short const*, long, int, void (CFRange, bool) block_pointer) + 1040
18 CoreText                       0xb5bfc DrawGlyphsAtPositions(TFont const*, unsigned short const*, CGPoint const*, unsigned long, CGContext*, (anonymous namespace)::DrawGlyphsConfiguration const&) + 4180
19 CoreText                       0x58798 CTFontDrawGlyphsAtPositionsInternal + 108
20 CoreText                       0x9ba0c TRun::DrawGlyphsAtPositionsInternal(CGContext*, CFRange, CGPoint const*, bool, TAttributes const*, bool) const + 728
21 CoreText                       0xae6d0 CTRunDraw + 188
22 HLLDriver-SEA                  0x154ad18 YYTextDrawRun + 2233 (YYTextLayout.m:2233)
23 HLLDriver-SEA                  0x15456b4 -[YYTextLayout drawInContext:size:point:view:layer:debug:cancel:] + 2618 (YYTextLayout.m:2618)
24 HLLDriver-SEA                  0x152d7b0 __30-[YYLabel newAsyncDisplayTask]_block_invoke.320 + 1135 (YYLabel.m:1135)
25 HLLDriver-SEA                  0x15302f4 -[YYTextAsyncLayer _displayAsync:] + 222 (YYTextAsyncLayer.m:222)
26 HLLDriver-SEA                  0x152fe24 -[YYTextAsyncLayer display] + 115 (YYTextAsyncLayer.m:115)
27 QuartzCore                     0x1c5d4 CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 408
28 QuartzCore                     0x2d0bc CA::Context::commit_transaction(CA::Transaction*, double, double*) + 464
29 QuartzCore                     0x6158c CA::Transaction::commit() + 652
30 QuartzCore                     0x49950 CA::Transaction::flush_as_runloop_observer(bool) + 84
31 CoreFoundation                 0x8ba50 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 32
32 CoreFoundation                 0x19bc4 __CFRunLoopDoObservers + 552
33 CoreFoundation                 0x77320 __CFRunLoopRun + 1020
34 CoreFoundation                 0x7bdfc CFRunLoopRunSpecific + 584
35 GraphicsServices               0x1984 GSEventRunModal + 160
36 UIKitCore                      0x376e7c -[UIApplication _run] + 868
37 UIKitCore                      0x376af4 UIApplicationMain + 312
38 HLLDriver-SEA                  0x432c main + 36 (main.m:36)
39 ???                            0x1d8a5ade0 (Missing)

```
